sdv/sampling/independent_sampler.py

```diff
@@ -0,0 +1,172 @@
+"""Independent Samplers."""
+import logging
+from copy import deepcopy
+
+import pandas as pd
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BaseIndependentSampler():
+    """Independent sampler mixin.
+
+    Args:
+        metadata (sdv.metadata.multi_table.MultiTableMetadata):
+            Multi-table metadata representing the data tables that this sampler will be used for.
+        table_synthesizers (dict):
+            Dictionary mapping each table to a synthesizer. Should be instantiated and passed to
+            sampler by the multi-table synthesizer using this sampler.
+        table_sizes (dict):
+            Dictionary mapping each table to its size. Should be instantiated and passed to the
+            sampler by the multi-table synthesizer using this sampler.
+    """
+
+    def __init__(self, metadata, table_synthesizers, table_sizes):
+        self.metadata = metadata
+        self._table_synthesizers = table_synthesizers
+        self._table_sizes = table_sizes
+
+    def _add_foreign_key_column(self, child_table, parent_table, child_name, parent_name):
```

Comment:
As we recently discussed, it seems like there can be multiple foreign keys to the same table. Maybe we should change the name here to make it plural and make the function expected to add all foreign key columns